gateway/api/management/commands/update_jobs_statuses.py

```diff
@@ -36,15 +38,37 @@ def handle(self, *args, **options):
                 else:
                     success = False
 
-                if not success:
-                    kill_ray_cluster(job.compute_resource.title)
-                    job.compute_resource.delete()
-                    job.compute_resource = None
-                    job_status = Job.FAILED
+                timeout = config.PROGRAM_TIMEOUT
+                if job.updated:
+                    endtime = job.updated + timedelta(days=timeout)
+                else:
```

Comment:
I think we can remove this else statement and put logic of stopping job under `if job.updated` clause. Since technically job can sit for N days in a queue and not started.